xwiki-rendering-macros/xwiki-rendering-macro-content/src/main/java/org/xwiki/rendering/internal/macro/content/ContentMacro.java

```diff
@@ -87,7 +90,19 @@ public boolean supportsInlineMode()
         throws MacroExecutionException
     {
         try {
-            return getSyntaxParser(parameters.getSyntax()).parse(new StringReader(content)).getChildren();
+            List<Block> blocks = getSyntaxParser(parameters.getSyntax()).parse(new StringReader(content)).getChildren();
+
+            // if the result is only one metadata block, we put the syntax metadata in it
+            if (blocks.size() == 1 && blocks.get(0) instanceof MetaDataBlock) {
```

Comment:
In practice there is almost no chance of this ever happening (you won't have that with all the parsers I know at least and I can't think of a reason for a parser to do that) so not sure it really worth it.

Reply:
Wasn't sure either and it's hard to test properly. I'll remove it.